### PR TITLE
feat: support snapcraft layout

### PR DESCRIPTION
**Pull request by PeerXu**
**If applied, this commit will...**

Goreleaser will support snapcraft layout.

bind to #2106 

**Why is this change being made?**

I am a goreleaser user.

I use goreleaser to build my own application: [meepo](https://github.com/PeerXu/meepo).

When try to build snap with strict confinement, service can not read configure files in isolated environment.

I figure out how to solve the problem, it should be use `layout`, for mapping files to sandbox.

But goreleaser not support snapcraft layout now, so I want it support the feature.

**# Provide links to any relevant tickets, URLs or other resources**

[Use layout](https://snapcraft.io/docs/snap-layouts)

### CODE REVIEW

**Comment by PeerXu** [comment]
It was so weird that relaunch the build action(close pr and reopen it over and over again), the test will pass at latest.

File: pkg/config/config.go [hunk 1]
```diff
@@ -439,23 +439,31 @@ type SnapcraftAppMetadata struct {
 	RestartCondition string `yaml:"restart_condition,omitempty"`
 }
 
+type SnapcraftLayoutMetadata struct {
+	Symlink  string `yaml:",omitempty"`
+	Bind     string `yaml:",omitempty"`
+	BindFile string `yaml:"bind-file,omitempty"`
```
Comment (caarlos0) [inline pkg/config/config.go:445]:
```suggestion
	BindFile string `yaml:"bindFile,omitempty"`
```

on goreleaser we always use snakeCase for config

File: www/docs/customization/snapcraft.md [hunk 2]
```diff
@@ -98,6 +98,19 @@ snapcrafts:
...
+    # With layouts, you can make elements in $SNAP, $SNAP_DATA, $SNAP_COMMON
+    # accessible from locations such as /usr, /var and /etc. This helps when using
+    # pre-compiled binaries and libraries that expect to find files and
+    # directories outside of locations referenced by $SNAP or $SNAP_DATA.
+    # More info about layout here:
+    # https://snapcraft.io/docs/snap-layouts
+    # More info about environment variables here:
+    # https://snapcraft.io/docs/security-sandboxing
+    # Default is empty.
+    layout:
```
Comment (caarlos0) [inline www/docs/customization/snapcraft.md:110]: can you document the other fields as well?

**Review by caarlos0 — COMMENTED** [review]
looking good so far, just a couple comments

File: pkg/config/config.go (hunk 1 above)
Comment (PeerXu) [reply pkg/config/config.go]: ok, change `bind-file` to `bind_file`

**Comment by PeerXu** [comment]
@caarlos0 Updated.

**Comment by caarlos0** [comment]
Thanks!

**Pull request merged**